src/commands/commerce/extension/register.ts

```diff
@@ -0,0 +1,140 @@
+
+    public static flagsConfig = {
+        'registered-extensionname': flags.string({
+            char: 'r',
+            description: msgs.getMessage('extension.register.regExtensionNameFlagDescription'),
+        }),
+        extensionpointname: flags.string({
+            char: 'e',
+            description: msgs.getMessage('extension.register.extensionPointNameDescription'),
+        }),
+        'apex-classname': flags.string({
+            char: 'a',
+            description: msgs.getMessage('extension.register.apexClassNameFlagDescription'),
+        }),
+        // optional flag
+        'apex-namespace': flags.string({
+            char: 'm',
+            description: msgs.getMessage('extension.register.apexNameSpaceFlagDescription'),
+        }),
+    };
+
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<unknown> {
+        this.ux.log(
+            `Retrieving and registering Apex Class ${
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                this.flags['apex-classname']
+            }' using username: ${this.org.getUsername()} \n..........`
+        );
+        return {
+            res: this.registerApex(
+                this.flags['registered-extensionname'],
+                this.flags['extensionpointname'],
+                this.flags['apex-classname'],
+                this.org.getUsername()
+            ),
+        };
+    }
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public registerApex(
+        storeRegisteredName: string,
+        storeEPN: string,
+        storeApexClass: string,
+        storeUserName: string
+    ): string {
+        // get apexClass id from table
+        const apexClassId = this.getApexClass(storeApexClass, storeUserName);
+
+        // checks if user passed valid EPN
+        const epnVal = forceDataSoql(
+            `SELECT Value FROM PicklistValueInfo WHERE Value='${storeEPN}' AND EntityParticle.DurableId = 'RegisteredExternalService.ExtensionPointName' LIMIT 1`,
+            storeUserName
+        );
+        if (epnVal.result.totalSize === 0) {
+            throw new SfdxError(msgs.getMessage('extension.register.errEPN'));
+        }
+
+        // Inserts 5 fields into registeredExternalService table
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'RegisteredExternalService',
+            `DeveloperName=${storeRegisteredName} MasterLabel=${storeRegisteredName} ExtensionPointName=${storeEPN} ExternalServiceProviderId=${apexClassId} ExternalServiceProviderType='Extension'`,
+            storeUserName
+        );
+        // Validates/checks for already existing unique name?
+        if (results instanceof SfdxError && results.message.indexOf('DUPLICATE_VALUE') > 0) {
```

Comment:
what if the result is an instanceof SfdxError but the message does not contain  'DUPLICATE_VALUE'. We need an else clause to handle any other error with a generic message something like -  `"Error while creating the RegisteredExternalService record: "+e.message`